Feature/npr

fjosw opened the pull request:
Replaced `sum` by `sum_large` in FourQuarkFullyConnected.hpp. Removed safeguarding for gpu compilation.

fjosw committed:
fix: sum in FourQuarkFullyConnected replaced by sumD_large

fjosw committed:
Removed enable-npr-fourquark macros

fjosw committed:
GRID_CUDA safeguard in FourQuarkFullyConnected removed.

fjosw committed:
sumD_large in FourQuarkFullyConnected.hpp replaced by sum_large

fjosw committed:
Hadrons_NPR_FOURQUARK removed from configure.ac

felixerben commented:
This is great, thanks Fabian! Just to make sure: Is this still supposed to be a draft or d you want me to merge it so that it enters the main pull request to Antonin's develop branch?

fjosw commented:
The changes are final from my side but the corresponding adjustments to Grid haven't made it to the `develop` branch of Grid yet so the CI would most certainly fail. I would propose to wait for the Grid changes to prevent further confusion.

fjosw committed:
Cleaned up comments and whitespaces

fjosw committed:
sum in Bilinear module changed to sum_large in order to suppress
warnings triggered on gpu machines on which PropagatorFields do not fit
in the shared memory.

fjosw commented:
Peter merged the gpu sum changes in to develop so we should be finally able to finish this pull request. I now also changed the `sum` in the bilinear module to `sum_large` to suppress warnings on gpu machines and did a few final cosmetic changes. This should be ready to be merged to the main PR.
